pkg/rebuild/maven/gradleinfer.go

```diff
@@ -36,7 +36,7 @@ func GradleInfer(ctx context.Context, t rebuild.Target, mux rebuild.RegistryMux,
 		ref = sourceJarGuess.Hash.String()
 		log.Printf("using source jar heuristic ref: %s", ref[:9])
 	default:
-		return nil, errors.Errorf("no valid git ref")
+		return nil, errors.Errorf("no git ref (gradle)")
```

Comment:
don't really enjoy this style. at that point you'd need all of the errors to annotate their origin. instead, you could wrap the higher-level consumer of this error with "gradle infer" or "inferring gradle" and test that interface (or just test this one and rely on the higher-level to differentiate between the two).

Reply:
A better idea indeed.